spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkRowLevelOperationsTestBase.java

```diff
@@ -255,7 +255,7 @@ protected <T> void createOrReplaceView(String name, List<T> data, Encoder<T> enc
   private Dataset<Row> toDS(String schema, String jsonData) {
     List<String> jsonRows =
         Arrays.stream(jsonData.split("\n"))
-            .filter(str -> !str.trim().isEmpty())
```

Comment:
Why this change?

Reply:
Also curious why we need to make the changes in v3.5? Can we focus on v4.0 change in this PR?